internal/pkg/term/progress/status.go

```diff
@@ -7,24 +7,49 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/aws/copilot-cli/internal/pkg/aws/cloudformation"
 	"github.com/aws/copilot-cli/internal/pkg/term/color"
 )
 
 var (
-	notStartedStackStatus = stackStatus{
-		value: "not started",
+	notStartedStackStatus = cfnStatus{
+		value: notStartedResult{},
 	}
 )
 
-type stackStatus struct {
-	value  cloudformation.StackStatus
+type result interface {
+	IsSuccessful() bool
+	IsFailure() bool
+	InProgress() bool
+	fmt.Stringer
+}
+
+// notStartedResult represents an unbegun status that implements the result interface.
+type notStartedResult struct{}
+
+func (r notStartedResult) IsSuccessful() bool {
```

Comment:
nit: comment pls

Reply:
Done! But recently I found out that the stdlib doesn't do this if the type isn't public for example:
https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/io/multi.go;l=17

Personally, I think it's okay if there is a comment on the type highlighting the interfaces that it implements 💭